### PR TITLE
[no ticket][risk=no] Puppeteer Reduce tests flakiness

### PR DESCRIPTION
**Pull request by aweng98**
Contains a number of small fixes:
1. Adding missing `{visible: true}` in `page.waitForXPath` function.
2. Increase timeout for long notebook tests. test playback time varies a lot, may take a long time to run.
3. Add retry in few flaky functions.
4. Rework `waitUntilEnabled` function.

`runtime-status-update.spec.ts` is flaky but the issue is not addressed in this PR.

### CODE REVIEW

File: e2e/app/element/button.ts [hunk 1]
```diff
@@ -27,14 +25,26 @@ export default class Button extends BaseElement {
...
   /**
    * Wait until button is clickable (enabled).
-   * @param {string} selector (Optional) Button Xpath selector.
+   * @param {string} xpathSelector (Optional) Button Xpath selector.
    * @throws Timeout exception if button is not enabled after waiting.
    */
-  async waitUntilEnabled(): Promise<void> {
-    const isCursorPointer = async(): Promise<boolean> => {
-      return await getPropValue<string>(await this.asElementHandle(), 'cursor') === 'pointer';
+  async waitUntilEnabled(xpathSelector?: string): Promise<void> {
```
Comment (aweng98) [inline e2e/app/element/button.ts:31]: Restore original code because `waitForFn` is not working well here.

File: e2e/app/element/checkbox.ts [hunk 2]
```diff
@@ -29,22 +28,22 @@ export default class Checkbox extends BaseElement {
    * Click (check) checkbox element.
    */
   async check(maxAttempts: number = 2): Promise<void> {
```
Comment (aweng98) [inline e2e/app/element/checkbox.ts:30]: Rewrite because `waitForFn` is not working well here.

File: e2e/utils/waits-utils.ts [hunk 3]
```diff
@@ -263,7 +263,7 @@ export async function waitForText(page: Page,
  */
 export async function waitWhileLoading(page: Page, timeout: number = 90000): Promise<void> {
   const notBlankPageSelector = '[data-test-id="sign-in-container"], title:not(empty), div.spinner, svg[viewBox]';
-  const spinElementsSelector = '[style*="running spin"], .spinner:empty, svg[style*="running rotation"]:not([aria-hidden="true"])';
+  const spinElementsSelector = '[style*="running spin"], .spinner:empty, [style*="running rotation"]:not([aria-hidden="true"])';
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:266]: In notebook runtime panel, there is a `img` (spinner) element. remove `svg` so it works for `img` spinner too.

File: e2e/utils/waits-utils.ts [hunk 4]
```diff
@@ -276,6 +276,8 @@ export async function waitWhileLoading(page: Page, timeout: number = 90000): Pro
     const elements = document.querySelectorAll(css);
     return elements && elements.length === 0;
   }, {polling: 'mutation', timeout}, spinElementsSelector);
+
+  await page.waitForTimeout(500);
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:280]: Little time to allow UI elements to start rendering in UI

File: e2e/package.json [hunk 5]
```diff
@@ -51,6 +51,7 @@
     "fs-extra": "9.0.1",
     "jest": "26.6.3",
     "jest-circus": "26.6.3",
+    "jest-config": "26.6.3",
```
Comment (aweng98) [inline e2e/package.json:54]: Missed missing package.

File: e2e/app/element/checkbox.ts [hunk 6]
```diff
@@ -29,22 +28,22 @@ export default class Checkbox extends BaseElement {
...
+      await this.clickWithEval();
+      await this.page.waitForTimeout(500);
+      const isChecked = await this.isChecked();
+      if (isChecked) {
+        return;
+      }
+      if (maxAttempts <= 0) {
+        return;
+      }
+    maxAttempts--;
```
Comment (jaycarlton) [inline e2e/app/element/checkbox.ts:41]: check inidentation here.

File: e2e/tests/notebook/notebook-download.spec.ts [hunk 7]
```diff
@@ -68,6 +68,6 @@ describe('Jupyter notebook download test', () => {
     // As of 9/25/20 I was unable to find a clear mechanism for accessing this.
 
     await notebook.deleteNotebook(notebookName);
-  }, 20 * 60 * 1000);
+  }, 30 * 60 * 1000);
```
Comment (jaycarlton) [inline e2e/tests/notebook/notebook-download.spec.ts:71]: 30 minutes? This is min * sec * millis riight?

**Review by jaycarlton — APPROVED** [review]
next time, splitting these up would help.

File: e2e/tests/notebook/notebook-download.spec.ts (hunk 7 above)
Comment (aweng98) [reply e2e/tests/notebook/notebook-download.spec.ts:71]: yes, it's 30 minutes.

**Review by aweng98 — COMMENTED** [review]
Fixed indentation. Thanks

**Pull request merged**